x-pack/legacy/plugins/infra/public/components/formatted_time.tsx

```diff
@@ -17,13 +17,33 @@ const getFormattedTime = (
   return userFormat ? moment(time).format(userFormat) : moment(time).format(fallbackFormat);
 };
 
-export const useFormattedTime = (time: number, fallbackFormat?: string) => {
-  const [dateFormat] = useKibanaUiSetting('dateFormat');
-  const formattedTime = useMemo(() => getFormattedTime(time, dateFormat, fallbackFormat), [
+interface UseFormattedTimeOptions {
+  show?: 'date' | 'time' | 'all';
+  fallbackFormat?: string;
+}
+
+export const useFormattedTime = (time: number, options: UseFormattedTimeOptions = {}) => {
```

Comment:
You can move the default values into the definition 
```suggestion
export const useFormattedTime = (time: number, options: UseFormattedTimeOptions = {show: 'all'}) => {
```

and skip copying them into `opts` in the next line

Reply:
I thought about that, but if the consumer of the function passes an `options` the default values get lost.

```ts
useFormattedTime(1234, { fallbackFormat: "YYYY-mm-dd" }); // <-- `options.show === undefined`
```

Reply:
Right. My bad, that'd be:

```suggestion
export const useFormattedTime = (time: number, {show='all', fallbackFormat}: UseFormattedTimeOptions = {}) => {
```

[quick example/test](https://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgKoGcIDED2UC2cYkAJgCrD4QDyADmMDiOsgN4BQyy6AFjgO4B+AFzIA5CSIQxyAD7iGVGfLFwANmrEBuTsngaARogDWuAkRHcwUUAHMdAX3bsETdGGQBXTGcLEI5JQoALzIABSKEKIgnvgG0AA0bLwCwaoaYkn6akYIpnh+DqIY2AVEpBRUdAxuyKGsDgCUdQB8bLquzDhqEAB0aji2Yawp-FnqOSa+RE3sTuzepeb+gVRhAIwATADMACyNCz5lK5UQGzu7SazZufnLomIAHsgAnsgAXmKzi9MnQed7K6jB6SSBfRpAA)

Reply:
Ohhh, TIL!

Reply:
That seems like a typo (syntax error). No worries. No need to get stuck on it now, but it should work. Can always come back to it.